src/DeribitSolution/Pipelines/Pipeline.cs

```diff
@@ -0,0 +1,35 @@
+﻿using ServiceClient.Abstractions;
+
+namespace ConsoleApp.Pipelines;
+
+internal abstract class Pipeline
+{
+    private readonly IServiceClient client;
+
+    protected Pipeline(IServiceClient client)
+    {
+        this.client = client;
+    }
+
+    public virtual async Task RunAsync(CancellationToken? cancellationToken = null)
```

Comment:
I think _CancellationToken_ is required and never _null_ because it will be used to request the exit from the loop